modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationProcessor.java

```diff
@@ -0,0 +1,600 @@
+                                new PartitionReconciliationValueMeta(
+                                    cacheObjOpt.get().valueBytes(ctx),
+                                    cacheObjOpt.map(o -> ConsistencyCheckUtils.objectStringView(ctx, o)).orElse(null),
+                                    uuidBasedEntry.getValue().version())
+                                :
+                                null);
+                    }
+
+                    KeyCacheObject key = entry.getKey().getKey();
+
+                    key.finishUnmarshal(ctx, null);
+
+                    RepairMeta repairMeta = entry.getValue();
+
+                    Optional<CacheObject> cacheObjRepairValOpt = Optional.ofNullable(repairMeta.value());
+
+                    res.add(
+                        new PartitionReconciliationDataRowMeta(
+                            new PartitionReconciliationKeyMeta(
+                                key.valueBytes(ctx),
+                                ConsistencyCheckUtils.objectStringView(ctx, key)),
+                            valMap,
+                            new PartitionReconciliationRepairMeta(
+                                repairMeta.fixed(),
+                                cacheObjRepairValOpt.isPresent() ?
+                                    new PartitionReconciliationValueMeta(
+                                        cacheObjRepairValOpt.get().valueBytes(ctx),
+                                        cacheObjRepairValOpt.map(o -> ConsistencyCheckUtils.objectStringView(ctx, o)).orElse(null),
+                                        null)
+                                    :
+                                    null,
+                                repairMeta.repairAlg())));
+                }
+
+                inconsistentKeys.computeIfAbsent(cacheName, k -> new HashMap<>())
+                    .computeIfAbsent(partId, k -> new ArrayList<>())
+                    .addAll(res);
+            }
+            catch (IgniteCheckedException e) {
+                log.error("Broken key can't be added to result. ", e);
+            }
+        }
+    }
+
+    /**
+     *
+     */
+    private PartitionReconciliationResult prepareResult() {
+        synchronized (inconsistentKeys) {
+            synchronized (skippedEntries) {
+                return new PartitionReconciliationResult(
+                    ignite.cluster().nodes().stream().collect(Collectors.toMap(
+                        ClusterNode::id,
+                        n -> n.consistentId().toString())),
+                    inconsistentKeys,
+                    skippedEntries
+                );
+            }
+        }
+    }
+
+    /**
+     * Reconciliation local progress tracker.
+     */
+    private class WorkProgress {
+        /** Work progress print interval. */
+        private final long WORK_PROGRESS_PRINT_INTERVAL = getLong("WORK_PROGRESS_PRINT_INTERVAL", 1000 * 60 * 3);
+
+        /**
+         *
+         */
+        private long total;
```

Comment:
Use one-line comment and please add minimal javadoc for each field in the class..